Core/src/org/sleuthkit/autopsy/communications/ContactNode.java

```diff
@@ -77,18 +78,61 @@ protected Sheet createSheet() {
             sheet.put(sheetSet);
         }
 
-        sheetSet.put(new NodeProperty<>("email", Bundle.ContactNode_Email(), "",
-                getAttributeDisplayString(artifact, TSK_EMAIL))); //NON-NLS
-        sheetSet.put(new NodeProperty<>("phone", Bundle.ContactNode_Phone(), "",
-                getAttributeDisplayString(artifact, TSK_PHONE_NUMBER))); //NON-NLS
-        sheetSet.put(new NodeProperty<>("mobile", Bundle.ContactNode_Mobile_Number(), "",
-                getAttributeDisplayString(artifact, TSK_PHONE_NUMBER_MOBILE))); //NON-NLS
-        sheetSet.put(new NodeProperty<>("home", Bundle.ContactNode_Home_Number(), "",
-                getAttributeDisplayString(artifact, TSK_PHONE_NUMBER_HOME))); //NON-NLS
-        sheetSet.put(new NodeProperty<>("office", Bundle.ContactNode_Office_Number(), "",
-                getAttributeDisplayString(artifact, TSK_PHONE_NUMBER_OFFICE))); //NON-NLS
-        sheetSet.put(new NodeProperty<>("url", Bundle.ContactNode_URL(), "",
-                getAttributeDisplayString(artifact, TSK_URL))); //NON-NLS
+        // Sorting the attributes by type so that the duplicates can be removed
+        // and they can be grouped by type for display.
+        try {
+            HashMap<String, BlackboardAttribute> phoneNumList = new HashMap<>();
+            HashMap<String, BlackboardAttribute> emailList = new HashMap<>();
+            HashMap<String, BlackboardAttribute> nameList = new HashMap<>();
+            HashMap<String, BlackboardAttribute> otherList = new HashMap<>();
+            for (BlackboardAttribute bba : artifact.getAttributes()) {
+                if (bba.getAttributeType().getTypeName().contains("TSK_PHONE")) {
+                    phoneNumList.put(bba.getDisplayString(), bba);
+                } else if (bba.getAttributeType().getTypeName().contains("TSK_EMAIL")) {
+                    emailList.put(bba.getDisplayString(), bba);
+                } else if (bba.getAttributeType().getTypeName().contains("TSK_NAME")) {
+                    nameList.put(bba.getDisplayString(), bba);
+                } else {
+                    otherList.put(bba.getDisplayString(), bba);
+                }
+            }
+            String propertyID = BlackboardAttribute.ATTRIBUTE_TYPE.TSK_NAME.getLabel();
+            int count = 0;
+            for (BlackboardAttribute bba : nameList.values()) {
```

Comment:
The following 3 "for" loops all look eerily similar differing only in the propertyID used and the collection they operate on. Perhaps these can be consolidated into a single method that does the work (taking the propertyID and the collection as parameters)?

Reply:
I thought about that.  I guess it would make it a bit cleaner.